osquery/core/watcher.cpp

```diff
@@ -359,17 +377,21 @@ bool WatcherRunner::watch(const PlatformProcess& child) const {
 }
 
 void WatcherRunner::stopChild(const PlatformProcess& child) const {
+  /* We leave 2 seconds for the rest of the logic to shutdown,
+   and 2 seconds for the forced kill to do the reaping */
```

Comment:
The flag's help says it's the time to wait for graceful, so this oughtn't substract force kill time from it.

Reply:
I have no opinion what correct is. But the text disagrees with the help comment. Not sure if there's a simple language tweak. This is nit level...

Reply:
The help comment in the alarm timeout flag is referring to the whole shutdown procedure, which means any process that osquery starts, any internal service that osquery runs.

The comment you're referring to is in the Watcher, in the code that is killing the child process. The forced kill is referred to the Watcher force killing the child process, which is the second attempt it's doing to stop the child, not the whole osquery package.

Would changing the comment to help a little clarifying things?:
```
/* We leave 2 seconds for the rest of the logic to shutdown, and 2 seconds for the forced kill of the child to do the reaping */
```

Reply:
I don't think so? The CLI flag says `Seconds to wait for a graceful shutdown.` but I think it's really "Seconds to allow for shutdown", since this also covers a force kill period

Reply:
oh I see what you mean now, you're right. Yes "Seconds to allow for shutdown" is correct.